app/controllers/concerns/idv/getting_started_ab_test_concern.rb

```diff
@@ -1,13 +1,22 @@
 module Idv
   module GettingStartedAbTestConcern
     def getting_started_a_b_test_bucket
-      AbTests::IDV_GETTING_STARTED.bucket(sp_session[:request_id] || session.id)
+      AbTests::IDV_GETTING_STARTED.bucket(current_user&.uuid)
```

Comment:
when would `current_user` be nil? does that mean we should be using `effective_user` or something else instead?

Reply:
`current_user` shouldn't be nil, but I'd rather put in a guard than get a 500 because someone came to the welcome step from the hybrid flow when they're not really logged in (and shouldn't be on the welcome step anyway). I'm not sure we get any benefit from using `effective_user`. Thoughts?

Reply:
I feel like... `current_user` is a key part of any logged-in action and so not having that indicates much bigger problems, and if it does cause a crash, that would be good

Reply:
I think we should drop the nil-safe operator. I'd rather raise a 500 and get a signal sooner.

Reply:
When someone starts on desktop and chooses hybrid flow and moves to their phone, and then also starts the flow on their phone, we end up in a state where they can visit IdV pages while not really logged in. I'm not sure why the usual `before_action :confirm_two_factor_authenticated` doesn't stop that, but we'd have to track that down before we stop putting guards in.

Reply:
that's what `effective_user` should be for, the mobile part of the hybrid flow where the user is logged in via slightly different session

Reply:
I had the impression we were moving away from using `effective_user`, but I'm wiling to put it in here if that's the preferred solution.

Reply:
if the intention is someone shouldn't be able to access this page in a hybrid session, then we should use `current_user` I think?

Reply:
My $0.02 is we should not be adding new references to `effective_user`. I'm fine with removing `&` here.

Reply:
I also hate `effective_user`.

The user will be nil and removing the `&` will be problematic here in each of the controllers in the `HybridMobile` namespace. There are three controllers in this PR where we are now calling this new method (here is a ref for what it looks like: https://github.com/18F/identity-idp/pull/8822/files#diff-f7a74e3e817f1c0a29e6561c1379f3023a6134c9cf909db32e4f535a2a61c0eeR59)

This is a little troublesome because I'm pretty sure we _do_ want to log information about which A/B test bucket the user is in in those cases. To add to the frustration, we know who the user is; they just aren't signed in. We can use `document_capture_user` to find them but that will `NoMethodError` outside of the controllers in the `HybridMobile` namespace.

This issue where things are slightly different in different contexts is the major weakness of concerns imo. I'm not sure how we work around it without having some sort of "analytics_user" construction that starts to get close to re-implementing effective user.

Reply:
Let me make a minor correct. [getting_started_ab_test_analytics_args](https://github.com/18F/identity-idp/pull/8822/files#diff-f17c20618db4bd80a5b1ef60d060869dc4bd90919f3cfe3c400c9be2ae26677bR13) already guards against a missing current user. As a result the hybrid mobile controllers will not log anything for the analytics args, but also won't result in an error.

Reply:
Resolved in https://github.com/18F/identity-idp/pull/8822/commits/545fd5780d6e367d3b21b132929016099dc209ee. First check if document_capture_user is available (hybrid_mobile flow), otherwise using current_user.uuid, without the nil-safe guard.